source/node_test.go

```diff
@@ -383,3 +420,98 @@ func testNodeSourceEndpoints(t *testing.T) {
 		})
 	}
 }
+
+func testNodeEndpointsWithIPv6(t *testing.T) {
+	for _, tc := range []struct {
+		title              string
+		annotationFilter   string
+		labelSelector      string
+		fqdnTemplate       string
+		nodeName           string
+		nodeAddresses      []v1.NodeAddress
+		labels             map[string]string
+		annotations        map[string]string
+		exposeInternalIPv6 bool // default to true for this version. Change later when the next minor version is released.
+		unschedulable      bool // default to false
+		expected           []*endpoint.Endpoint
+		expectError        bool
+	}{
+		{
+			title:              "node with only internal IPs with expose internal IP as false shouldn't return AAAA endpoints with internal IPs",
+			nodeName:           "node1",
+			exposeInternalIPv6: false,
+			nodeAddresses:      []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: "2.3.4.5"}, {Type: v1.NodeInternalIP, Address: "2001:DB8::9"}},
+			expected: []*endpoint.Endpoint{
+				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"2.3.4.5"}},
+			},
+		},
+		{
+			title:              "node with both external, internal, and IPv6 IP returns endpoints with external IPs",
```

Comment:
This title does not match the test case. The node in the test only has `ExternalIP`s.